test/backend/test_signature.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import absolute_import
```

Comment:
Thanks for adding a signature test, this is very helpful. Please add the copyright when adding a new file, and I would urge you to add a paragraph of comments in the file describing the test, to make it easier for us to expand its functionality in the future if required.

Reply:
Is it possible to incorporate this into test.py? Most of the code here is copied from test.py. Maintaining two copies would be error prone. If test.py gets too large, we can consider modularize it but we don't want to maintain two copies of similar code.

BTW, ID shiyz818 has been added to the whitelist to automatically trigger Jenkins builds.

Reply:
Hi Zhen Yu, the current pattern is that different tests can be selected by either specifying an option or set an env var. Yes I do realize that signature tests do not need to go through onnx backend. But I think we can still extend test.py with for example --signature and TEST_SIGNATURE to branch to your code.

IMO test.py is a bit disorganized in its current form. So if you can take this opportunity to modularize and organize it a bit that would be very helpful. Thanks.

Reply:
I see. Let me try it.

Reply:
Thanks Zhen Yu. Just be aware that in #985 I made some changes to test.py to add the JNI backend tests. So you probably want to make your changes on top of that instead.

Reply:
Hi @gongsu832. I incorporate signature test into test.py framework. Here is main changes:
* global variable initialization in varaibles.py
* common function `compile_model` in common.py
* Class SignatureBackendTest is inherited from onnx.backend.test.BackendTest for signature testing. It overwrites `_add_model_test` and `assert_similar_outputs`

Would you help to review? I will wait #985 be merged and then add JNI part. Thanks!

Reply:
@shizy818 Thanks Zhen Yu for the changes. From a high level I think the pieces look good! There are errors in the code (e.g., ExecutionSession now has 3 params but still called with two) but I think that's only because you are still waiting for #985. This should make the backend tests a lot more maintainable. Thanks!

Reply:
@shizy818 @gongsu832 seems that #985 is merged now, so is this one good to go?

Reply:
I think @shizy818 is still in the process of finishing up the modularization and cleanup of test.py so let's wait for him to answer.

Reply:
Hi @gongsu832 I merged JniExecutionSession from #985 into inference_backend.py last week. Logically I guess it is what it looks now. I double check code cleanup today. Is it Ok for you?

Reply:
Hi @shizy818 yes it looks good. Thanks for fixing up test.py.